File: src/main/java/fi/metatavu/metaform/server/persistence/dao/AttachmentReplyFieldDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.AttachmentReplyField;
import fi.metatavu.metaform.server.persistence.model.Reply;

/**
 * DAO class for AttachmentReplyField entity
 * 
 * @author Antti Leppä
 */
@ApplicationScoped
public class AttachmentReplyFieldDAO extends ReplyFieldDAO<AttachmentReplyField> {
  
  /**
   * Creates new AttachmentReplyField 
   * 
   * @param id id
   * @param reply reply
   * @param name field name
   * @param value field value
   * @return created field
   */
  public AttachmentReplyField create(UUID id, Reply reply, String name) {
    AttachmentReplyField replyField = new AttachmentReplyField(); 
    replyField.setId(id);
    replyField.setName(name);
    replyField.setReply(reply);
    return persist(replyField);
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/dao/AttachmentReplyFieldDAO.kt
package fi.metatavu.metaform.server.persistence.dao

import fi.metatavu.metaform.server.persistence.model.AttachmentReplyField
import fi.metatavu.metaform.server.persistence.model.Reply
import java.util.*
import javax.enterprise.context.ApplicationScoped

/**
 * DAO class for AttachmentReplyField entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class AttachmentReplyFieldDAO : ReplyFieldDAO<AttachmentReplyField>() {
  /**
   * Creates new AttachmentReplyField
   *
   * @param id id
   * @param reply reply
   * @param name field name
   * @return created field
   */
  fun create(id: UUID?, reply: Reply, name: String): AttachmentReplyField? {
    val replyField = AttachmentReplyField()
    replyField.id = id
    replyField.name = name
    replyField.reply = reply
    return persist(replyField)
  }
}